### ISSUE TITLE
 app rather large

### ISSUE THREAD

**Issue by jidanni** [kwattsorg/adbshellkit#24]
Why is this app 250 megabytes big?

Shouldn't it be just 2.5 megabytes big?

**Comment by jidanni** [comment]
oh, I see. It's not the APK size I was looking at. It was the total memory size on my device. But anyway yeah, why does it need to store 250 megabytes?

**Comment by kwatkins** [comment]
The added binaries and tools and having them pre-compiled for each
supported platform (arm, x86, etc)

The location of the binaries changed between versions, so if you had an
older version those binaries likely weren't removed. This is mostly fixed
now.

All that being said, this app could totally use a better system of getting
the binaries. Something where the install includes a core set of binaries
and rest can be downloaded from cloud on-demand and/or when that command is
executed by the user. Open to any and all ideas around this!
-- Kevin

**Comment by jidanni** [comment]
I know google play lets you have different apks for different
architectures, so maybe only include the binary for the relevant
architecture, if that's not what you do already.

Also perhaps have new versions clean up all the mess if it finds any,
upon installation.

**Comment by kwatkins** [comment]
You are totally right on this. Currently binaries are bundled into one APK and the app has gotten so large it's likely throwing up a "need wifi to download" type of adoption barrier. 

Tasks for this issue, and it'll likely come in phases, 
- Trim binaries to the "core", for commands beyond that, bundle inside another zip/package that will get installed on-demand or if a command is run that requires it. Add another field for commands containing the url/location of the binary with integrity hashes. If there is already a tool out there that does this, even better. 
- Update build/release to create apks for different architectures and include only the binaries and libs for that architecture. When getting the extra tools, we need to make sure an architecture check is done and pulls down the binaries for that platform.
